lib/text/listener.js

```diff
@@ -83,6 +83,82 @@ class Importer extends SHRParserListener {
     }
   }
 
+    enterValuesetDefsDoc(ctx) {
+        const ns = ctx.valuesetDefsHeader().namespace().getText();
+        this._currentNs = ns;
+        if (typeof this._nsMap[ns] == 'undefined') {
+            this._nsMap[ns] = new Namespace(ns);
+        }
+    }
+
+    exitValuesetDefsDoc(ctx) {
+        // clear current namespace and uses namespaces
+        this._currentNs = '';
+        this._usesNs = [];
+    }
+
+    enterValuesetDef(ctx) {
+        const name = ctx.valuesetHeader().simpleName().getText();
+        const id = new Identifier(this._currentNs, name);
+        const valueset = new Valueset(id);
+
+        var path = this._preprocessedData.resolvePath('default', this._currentNs);
+
+        valueset.url = path.url + name;
+        valueset.name = name;
+
+        this._currentDef = valueset;
+    }
+
+    exitValuesetDef(ctx) {
+      this.pushCurrentDefinition();
+    }
+
+    enterValuesetValue(ctx) {
+        const code = ctx.getText();
+        var value;
+        var result;
+        if (ctx.valuesetInlineValue() != null) {
+            result = this.processInlineValue(ctx.valuesetInlineValue());
+        } else if (ctx.fullyQualifiedCode() != null){   // e.g. MTH#1234567
+            result = this.processFullyQualifiedCode(ctx.fullyQualifiedCode());
+        }
+        this._currentDef.addCode(result);
+    }
+
+    enterValuesetProp(ctx) {
+        var result = this.processFullyQualifiedCode(ctx.conceptProp().concepts().fullyQualifiedCode()[0]);
```

Comment:
I'm pretty sure this will throw an error if a value set is defined with a `Description:` but without a `Concept:`.  This is because the `Description:` will result in `enterValuesetProp(ctx)` being called, but `ctx.conceptProp()` will be _null_.